source/vecs/signal.d

```diff
@@ -0,0 +1,84 @@
+module vecs.signal;
+
+version(vecs_unittest) import aurorafw.unit.assertion;
+
+struct Signal(T ...)
+{
+	alias SlotType = void delegate(T);
+	alias SlotTypeRef = void delegate(ref T);
+
+	@safe pure nothrow
+	void connect(SlotType slot)
+	{
+		slots ~= slot;
+	}
+
+	@safe pure nothrow
+	void disconnect(SlotType slot)
+	{
+		import std.algorithm : countUntil;
+		auto index = slots.countUntil(slot);
+		if (index != -1)
+		{
+			slots = slots[0 .. index] ~ slots[index+1 .. $];
```

Comment:
Do you plan to use containers here? You may allocate here and on the `connect()` call as well.

Reply:
I'm not using containers nor will be in the near future. VECS it's using GC all over at the moment so this is ok.